### ISSUE TITLE
Incorrect event path for arrays containing a mix of native and Y types

### ISSUE THREAD

**Issue by aartrost** [yjs/yjs#457]
**Checklist**

* [x] Are you reporting a bug? Use github issues for bug reports and feature requests. For general questions, please use https://discuss.yjs.dev/
* [x] Try to report your issue in the correct repository. Yjs consists of many modules. When in doubt, report it to https://github.com/yjs/yjs/issues/

**Describe the bug**
When you `observeDeep` a Map containing an array with a mix of native JS types & Y types the event's path will point to the incorrect index inside the Array. The problem does not occur if the array is only containing Y types.

**To Reproduce**
```
import * as Y from "yjs";

const doc = new Y.Doc();
const map = doc.getMap("t" + Date.now());

// Create a field with the array as value
map.set('my-array', new Y.Array());

// Fill the array with some strings and our Map
map.get('my-array').push(['a', 'b', 'c', new Y.Map()]);

// log the event path 
map.observeDeep((events) => console.log(events[0].path));

// set a value on the map inside of our array
map.get('my-array').get(3).set('hello', 'world');

// The event path that's logged will be ['my-array', 1] which is 'b'. The correct path
// should by ['my-array', 3]
```

**Expected behavior**
The resulting path should be `['my-array', 3]`, not `['my-array', 1]`

**Environment Information**
 - Chrome
 - Yjs @ v13.5.41

**Comment by krabhi1** [comment]
when will be this issue fixed?

**Comment by dmonad** [comment]
Thanks for the bug report @aartrost! Sorry it took so long..

The fix is released in Yjs@13.6.16
